rpki-validator-app/src/main/scala/net/ripe/rpki/validator/controllers/ExportController.scala

```diff
@@ -71,4 +75,65 @@ trait ExportController extends ApplicationController {
     response.getWriter.write(compact(render(("roas" -> roas))))
   }
 
+  get("/export.rpsl") {
+
+    contentType = "text/plain"
+    response.addHeader("Pragma", "public")
+    response.addHeader("Cache-Control", "no-cache")
+
+    val routes = new StringBuilder
+    getRtrPrefixes.map { rtr =>
+
+      val caName = if(rtr.trustAnchorLocator.isEmpty) "UNKNOWN" else rtr.trustAnchorLocator.get.getCaName.replace(' ', '-').toUpperCase
+      val dateTime = ISODateTimeFormat.dateTimeNoMillis().withZoneUTC().print(DateTime.now)
+
+      val allowedRoute = new AllowedRoute(rtr.asn, rtr.prefix, rtr.maxPrefixLength.getOrElse(rtr.prefix.getPrefixLength))
+      val possibleRoutes = getAllRoutesFor(allowedRoute.getPrefix, allowedRoute.getMaximumLength)
+
+      possibleRoutes.foreach { range =>
+        val version = if(IpResourceType.IPv6 == range.getType) "6" else ""
+
+        routes ++= s"""
```

Comment:
Should that be response.getWriter.write(s"....") instead to not to allocate a big string in memory?